### ISSUE TITLE
Pretraining networks for PPO

### ISSUE THREAD

**Issue by mhueneberg** [reinforceio/tensorforce#355]
Hello,
Firstly I have to say that you created an amazing Framework, thank you for that!
I am trying to pretrain the policy network for the PPO algorithm, because it converges very slowly for my data. I would like to use a pretrained policy network, for which PPO only does some small optimizations.  What I tried was basically something like this:
```
fetches = [self.agent.model.optimization, self.agent.model.loss_per_instance] 
feed_dict = {self.agent.model.states_input['state'] : batch_x,                                                                
                             self.agent.model.actions_input['action'] : batch_y,                                                               
                             self.agent.model.reward_input :  np.ones((train_batch_size,), dtype=np.float32)                                                                          
                             self.agent.model.terminal_input : np.zeros((train_batch_size,), dtype=np.bool_)                                                                            
                             self.agent.model.deterministic_input : True,                                                            
                             self.agent.model.update_input : True}
self.agent.model.session.run(fetches=fetches, feed_dict=feed_dict) 
```  
Unfortunately this does only seem to update the baseline network, which is not what I want. I have to say that it is hard for me to get a good overview over the whole tensorflow graph. So maybe someone can help me with some pointers on how I can train the policy network specifically?

Regards,
Mark

**Comment by michaelschaarschmidt** [comment]
Hi,
there is currently no pretraining method for policy gradient algorithms. If you look at the DQFD agentmodel, you see an example of how one can add an API method to pretrain from supervised data (in the pretrain method). 

The memory model (which PPO model inherits from) has a method to import experience:
https://github.com/reinforceio/tensorforce/blob/master/tensorforce/models/memory_model.py#L546

So what you would need to do is implement a method in the memory model that  performs an update after importing this data.

**Comment by michaelschaarschmidt** [comment]
Note that you can now do supervised pretraining and load just the relevant layers, see in the FAQ how

**Comment by Lingermania** [comment]
I have a large dataset of expert demonstrations and am wondering how to pre-train the PPO agent on this data. As I understand the post above it is not enough to import the experiences directly, I looked at the FAQ for examples but did not find any.

Any directions to further examples would be greatly appreciated.

Regards,
Kris

**Comment by michaelschaarschmidt** [comment]
Two ways:
1. See faq on just using supervised training externally, load in layers - works
2. One would need an update function that can be called independently. You can import memories already, but you would then need to be able to call update. It is not that difficult to just add this function (normally an update only happens in conjunction with an observe), so you can try that. I might do this some time soon myself for my research but will not promise anything

**Comment by Lingermania** [comment]
Thanks for a super quick response!
Importing experience requires me to import internals, for now, I have left this blank (an empty list) and I assume that is okay? Also, each experience I load should be a single episode, right?

**Comment by michaelschaarschmidt** [comment]
Just follow the example in the dqfd agent test and agent to look at the data layouting/internals
